provisioning/device/samples/solutions/BestPracticeSampleX509/BestPracticeSampleX509.cs

```diff
@@ -0,0 +1,533 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.Azure.Devices.Client.Samples;
+using Microsoft.Azure.Devices.Provisioning.Client.Transport;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Samples
+{
+    /// <summary>
+    /// Demonstrates how to register a device with the device provisioning service using a X.509 leaf certificate, and then
+    /// use the registration information to authenticate to IoT Hub.
+    /// </summary>
+    internal class BestPracticeSampleX509
```

Comment:
This file combined X509Sample and DeviceReconnectionSample.

Reply:
Meaning we got rid of the device connection sample, or just used it as a base?

Reply:
Maybe we should have the design reviewed to determine what feature we should include.